src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -399,10 +399,13 @@ private void SendFileInternal(string? fileName, ReadOnlySpan<byte> preBuffer, Re
         {
             // Open the file, if any
             SafeFileHandle? fileHandle = OpenFileHandle(fileName);
-
             SocketError errorCode;
-            // This can throw ObjectDisposedException.
-            errorCode = SocketPal.SendFile(_handle, fileHandle, preBuffer, postBuffer, flags);
+
+            using (fileHandle)
+            {
+                // This can throw ObjectDisposedException.
+                errorCode = SocketPal.SendFile(_handle, fileHandle, preBuffer, postBuffer, flags);
+            }
```

Comment:
This can be:
```C#
SocketError errorCode;
using (SafeFileHandle? fileHandle = OpenFileHandle(fileName))
{
    // This can throw ObjectDisposedException.
    errorCode = SocketPal.SendFile(_handle, fileHandle, preBuffer, postBuffer, flags);
}
```